multy_test/test_bitcoin_transaction.cpp

```diff
@@ -1369,3 +1369,47 @@ GTEST_TEST(BitcoinTransactionTest, transaction_set_destination_address_mainnet)
     EXPECT_ERROR(properties_set_string_value(&destination_mainnet, "address", "mzqiDnETWkunRDZxjUQ34JzN1LDevh5DpU"));
     EXPECT_ERROR(properties_set_string_value(&destination_mainnet, "address", "TEST"));
 }
+
+GTEST_TEST(BitcoinTransactionTest, SmokeTest_mainnet_dust_outputs)
+{
+    AccountPtr account = make_account(BLOCKCHAIN_BITCOIN, "5KDejYL1XnGkhwhH1ubSqiFCqXGxYzuJYHXhU7UqeVLEDeqBJ22");
+
+    const BigInt available(9809);
+    const BigInt dust_sent(500);
+    const BigInt for_dust_change(8500);
+
+    const TransactionTemplate TEST_TX
+    {
+        nullptr,
+        TransactionFee
+        { // fee:
+            BigInt{3}
+        },
+        { // Sources
+            {
+                available,
+                from_hex("63ebc8dd4de286a1ab32029f1986876ac981b8b9ccd597be3f2d6ac1961e02ee"),
+                0,
+                from_hex("76a914dda3945cee6bb677de3b090db38ef3053fc45acb88ac"),
+                nullptr
+            }
+        }
+    };
+    TransactionPtr transaction = make_transaction_from_template(TEST_TX, account, account->get_private_key());
+
+    Properties& send_pro = transaction->add_destination();
+    send_pro.set_property_value("address", "1MCvJ6pqJrGJEjo55RhLhr1de2wFLoDBXF");
+    send_pro.set_property_value("amount", dust_sent);
+
+    Properties& change = transaction->add_destination();
+    change.set_property_value("address", "1Q46s1EJXDMvVTCVQGKfrjEudvgu39uXu5");
+    change.set_property_value("is_change", 1);
+
+    EXPECT_ERROR(transaction_update(transaction.get()));
+
+    send_pro.set_property_value("amount", for_dust_change);
```

Comment:
Add comment:
Now change amount is dust, hence change amount should be implicitly set to 0 by TX itself.